src/ModelIO/MDLAsset.cs

```diff
@@ -1,12 +1,23 @@
 #if XAMCORE_2_0 || !MONOMAC
 using System;
+using XamCore.ObjCRuntime;
 namespace XamCore.ModelIO {
 	public partial class MDLAsset {
 		public MDLObject this [nuint index] {
 			get {
 				return GetObject (index);
 			}
 		}
+
+		[iOS (10,3), TV (10,2), Mac (10,12,4)]
+		public IMDLComponent this [Protocol key] {
+			get {
+				return ObjectForKeyedSubscript (key);
+			}
+			set {
+				SetObject (value, key);
```

Comment:
that's confusing because your new `SetObject` (even if `internal`) is not symmetric with the existing `GetObject`